src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -8694,16 +8697,56 @@ void ExportMusicXml::harmony(Harmony const* const h, FretDiagram const* const fd
         const String textName = h->hTextName();
         switch (h->harmonyType()) {
         case HarmonyType::NASHVILLE: {
-            m_xml.tag("function", h->hFunction());
-            m_xml.tag("kind", { { "text", textName } }, "none");
+            String alter;
+            String functionText = h->hFunction();
+            if (functionText.empty()) {
+                // we just dump the text as deprecated function
+                m_xml.tag("function", textName);
+                m_xml.tag("kind", "none");
+                break;
+            } else if (!functionText.at(0).isDigit()) {
+                alter = functionText.at(0);
+                functionText = functionText.at(1);
+            }
+            m_xml.startElement("numeral");
+            m_xml.tag("numeral-root", functionText);
+            if (alter == u"b") {
+                m_xml.tag("numeral-alter", "-1");
+            } else if (alter == u"#") {
+                m_xml.tag("numeral-alter", "1");
+            }
+            m_xml.endElement();
+            if (!h->xmlKind().isEmpty()) {
+                String s = u"kind";
+                String kindText = h->musicXmlText();
+                if (h->musicXmlText() != "") {
```

Comment:
why not `if (!h->musicXmlText().empty()) {`?

Reply:
changed this one and some more cases